SingularityService/src/main/java/com/hubspot/singularity/data/history/BlendedHistoryHelper.java

```diff
@@ -25,6 +27,10 @@
   protected abstract List<T> getFromZk(Q id);
   protected abstract List<T> getFromHistory(Q id, int historyStart, int numFromHistory);
 
+  protected Optional<Integer> getTotalCount(Q id) {
+    throw new NotImplementedException();
+  }
```

Comment:
Once this method is implemented in all the subclasses, this could be changed to `abstract` like the others.